### PR TITLE
Preserve current working directory when parsing PCM section

### PR DESCRIPTION
**Pull request by robertodr**
## Description
Parsing the PCM section of the input requires changing directories. As noted in #817 this is not really documented anywhere and affects users.

## Todos
Notable points that this PR has either accomplished or will accomplish.
* **User-Facing for Release Notes**
  - [x] CWD is saved before parsing PCM section and restored at the end.

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by loriab** [comment]
Did you happen to confirm `psi4 --test` is still ok, @robertodr? That runs PCM in PsiAPI mode.

**Comment by dgasmith** [comment]
Kinda odd that would break API mode.

**Comment by robertodr** [comment]
Trying to understand why it happens.

**Comment by loriab** [comment]
Everything is prone to breaking is PsiAPI mode because there's none of the saving new instance of psi4 between calculations. Try commenting out every test besides the PCM in `tests/pytest/test_addons.py` and see if that's ok? Note that `psi4 --test` is just `pytest -rs -v` behind the scenes.

**Comment by robertodr** [comment]
@raimis and/or @j3mdamas, can you provide an example where you are affected?

**Comment by j3mdamas** [comment]
@raimis?

**Comment by dgasmith** [comment]
@robertodr I think if you just run a PCM computation and print the CWD at the end. If this doesn't match the launching directory you have a problem, otherwise its fine.

**Comment by dgasmith** [comment]
Any update on whats wrong with this?

**Comment by robertodr** [comment]
`psi4 --test` is now OK. @j3mdamas @raimis issue #817 should be fixed.

**Comment by raimis** [comment]
@robertodr thanks for your effort!

File: psi4/driver/p4util/python_helpers.py [hunk 1]
```diff
@@ -225,14 +225,31 @@ def set_module_options(module, options_dict):
...
+    block: multiline string with PCM input in PCMSolver syntax.
+
+    Notes
+    -----
+    !Warning! This function changes directory from launch site to scratch and
+    back. This is needed because PCMSolver does not have a notion of an
+    absolute paths for its input file. If you are looking at this function as
+    an example, remember that changing directories is usually a **bad idea**
+    and **should not be done**.
+    """
```
Comment (andysim) [inline psi4/driver/p4util/python_helpers.py]: +1 for the warning here

Comment (robertodr) [reply psi4/driver/p4util/python_helpers.py]: @dgasmith warned me to add a warning 😺

File: external/upstream/pcmsolver/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
-if(${ENABLE_PCMSolver})
-    find_package(PCMSolver 1.1.10 CONFIG QUIET)
+    if(${ENABLE_PCMSolver})
+    find_package(PCMSolver 1.1.11 CONFIG QUIET)
```
Comment (loriab) [inline external/upstream/pcmsolver/CMakeLists.txt]: @robertodr, is 1.1.11 required or is 1.1.10 still compatible? if the former, all good. if the latter, then switch this and the psi4/CM one back to 10, please, while keeping the GIT_TAG one at 11. that way things won't break unnecessarily. thanks!

Comment (robertodr) [reply external/upstream/pcmsolver/CMakeLists.txt]: 1.1.10 is still compatible. Fixed according to your comment.

**Comment by j3mdamas** [comment]
thanks @robertodr and all. In which version will this be available? @raimis, we should change our code when it's available.

**Comment by robertodr** [comment]
It's on master already, you can just clone from github.

**Comment by j3mdamas** [comment]
Sorry, I wasn't clear. I meant on conda, on the main channel :) On 1.2?

**Comment by dgasmith** [comment]
Hmm, main channel will be 1.2. It's possible that you can get it from the dev conda channel soon. Would that work?

**Comment by j3mdamas** [comment]
We distribute mainly from the main. We can wait, no problem.

**Comment by loriab** [comment]
Sometime, could you take another look at `psi4 --test`, @robertodr? Whenever I run, I get the below, and it looks like it's happening outside the pcm_helper fn. That is, in c-side SCF where `@pcmsolver.inp` can't be read b/c the working dir wasn't left in the pcm subdir where `pcmsolver.inp`-->`@pcmsolver.inp`. 

Note that the changes in this PR would have only influenced psiapi mode, not psithon, since most of the pcm_helper fn is just repeated from inputparser. I'm not sure whether @raimis is trying to run psithon or psiapi — looks like the latter in intent and the former in practice. Tracing back to [here](https://github.com/Acellera/htmd/pull/409#discussion_r139685402), I think they're still psithon, so this PR may have targeted the wrong approach. Or I may have all my pcmsolver input redirects quite confused.

This being the fourth package that has conspired against conda, it may be one too many.

```
RHF-PCM, total algorithm
Error: Error: Getkw,  line 33 in  /scratch/psilocaluser/conda-builds/pcmsolver_1499412893611/work/src/utils/getkw/Getkw.cpp: Open failed: @pcmsolver.inp
```

```
        print('RHF-PCM, total algorithm')
>       energy_scf1, wfn1 = psi4.energy('scf', return_wfn=True)

../../_b_env_placehold_placehold/lib/python3.6/site-packages/psi4/tests/test_addons.py:425: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../_b_env_placehold_placehold/lib/python3.6/site-packages/psi4/driver/driver.py:460: in energy
    wfn = procedures['energy'][lowername](lowername, molecule=molecule, **kwargs)
../../_b_env_placehold_placehold/lib/python3.6/site-packages/psi4/driver/procrouting/proc.py:1996: in run_scf
    scf_wfn = scf_helper(name, post_scf=False, **kwargs)
../../_b_env_placehold_placehold/lib/python3.6/site-packages/psi4/driver/procrouting/proc.py:1271: in scf_helper
    scf_wfn = scf_wavefunction_factory(name, base_wfn, core.get_option('SCF', 'REFERENCE'))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

name = 'scf', ref_wfn = <psi4.core.Wavefunction object at 0x7fa9d0ef4af0>, reference = 'RHF'

    def scf_wavefunction_factory(name, ref_wfn, reference):
        """Builds the correct wavefunction from the provided information
        """
    
        if core.has_option_changed("SCF", "DFT_DISPERSION_PARAMETERS"):
            modified_disp_params = core.get_option("SCF", "DFT_DISPERSION_PARAMETERS")
        else:
            modified_disp_params = None
    
        # Figure out functional
        superfunc, disp_type = dft_funcs.build_superfunctional(name, (reference in ["RKS", "RHF"]))
    
        # Build the wavefunction
        core.prepare_options_for_module("SCF")
        if reference in ["RHF", "RKS"]:
>           wfn = core.RHF(ref_wfn, superfunc)
E           RuntimeError: std::exception

../../_b_env_placehold_placehold/lib/python3.6/site-packages/psi4/driver/procrouting/proc.py:1009: RuntimeError
```

**Comment by j3mdamas** [comment]
Hi @loriab,

Thanks for looking carefully into this. I don't really understand the terms psithon and psiapi you use, but our use of psi4 is always through the command-line tool, and this is the functionality where we were trying to fix the pcm capabilities, because we were rather using an ugly fix of saving the working directory and changing back to it in the end.

@raimis and I are available to provide any help you may need.

Happy holidays to the PSI4 team!

**Pull request merged**